packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -0,0 +1,829 @@
+    ) internal pure {
+        if (params.originator == address(0) || params.dstContract == address(0))
+            revert InvalidParams();
+    }
+
+    /**
+     * @dev Internal function to validate send message parameters
+     * @param params Parameters to validate
+     */
+    function _validateSendMessageParams(
+        BridgeTypes.ExecuteSendMessageParams calldata params
+    ) internal pure {
+        if (params.recipient == address(0) || params.originator == address(0))
+            revert InvalidParams();
+    }
+
+    /**
+     * @dev Internal function to validate if an adapter supports a specific operation type
+     * @param adapter The adapter address to validate
+     * @param operationType The type of operation to check support for
+     */
+    function _validateAdapterSupportsOperation(
+        address adapter,
+        BridgeTypes.OperationType operationType
+    ) internal view {
+        if (adapter == address(0)) revert NoSuitableAdapter();
+        if (!IBridgeAdapter(adapter).supportsOperation(operationType)) {
+            revert UnsupportedAdapterOperation();
+        }
+    }
+
+    /**
+     * @dev Internal function to validate provided fee against required fee
+     * @param providedFee The fee provided with the transaction
+     * @param requiredFee The required fee for the operation
+     */
+    function _validateFee(
+        uint256 providedFee,
+        uint256 requiredFee
+    ) internal pure {
+        if (providedFee < requiredFee) revert InsufficientFee();
+    }
+
+    /**
+     * @dev Internal function to handle refunds safely
+     * @param recipient Address to receive the refund
+     * @param amount Amount to refund
+     */
+    function _refund(address recipient, uint256 amount) internal {
+        if (amount > 0) {
+            (bool success, ) = recipient.call{value: amount}("");
+            if (!success) revert TransferFailed();
+        }
+    }
+
+    /**
+     * @dev Internal function to generate a unique operation ID and set initial status
+     * @param operationType Type of operation being performed
+     * @param destinationChainId Target chain ID
+     * @param asset Asset address (address(0) for non-asset operations)
+     * @param amount Amount (0 for non-asset operations)
+     * @param recipient Recipient address
+     * @param additionalData Additional data for ID generation (contract address, selector, etc.)
+     * @return operationId The generated operation ID
+     */
+    function _generateOperationId(
+        BridgeTypes.OperationType operationType,
+        uint16 destinationChainId,
+        address asset,
+        uint256 amount,
+        address recipient,
+        bytes memory additionalData
+    ) internal returns (bytes32 operationId) {
+        // Use nonce for better uniqueness and collision resistance
+        uint256 currentNonce = _useNonce(address(this));
+
+        operationId = keccak256(
+            abi.encode(
```

Comment:
not sure if it woul revert elswhere, but this could potentially allow sending two transactions with same params in the same block